src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static updateAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    Animal.alive.push(this);
```

Comment:
The `Animal.alive.push(this);` line is redundant here because the `Animal` constructor already adds the instance to the `alive` array. Remove this line to avoid adding the same instance twice.

Reply:
> The `Animal.alive.push(this);` line is redundant here because the `Animal` constructor already adds the instance to the `alive` array. Remove this line to avoid adding the same instance twice.

Good point, and actually you don't need `constructor` at all if you don't add any new properties or do other actions